pkg/alertmanager/api_grafana.go

```diff
@@ -316,6 +316,7 @@ func (am *MultitenantAlertmanager) SetUserGrafanaConfig(w http.ResponseWriter, r
 		util.WriteJSONResponse(w, errorResult{Status: statusError, Error: fmt.Sprintf("%s: %s", errReadingGrafanaConfig, err.Error())})
 		return
 	}
+	am.multitenantMetrics.grafanaConfigSize.WithLabelValues(userID).Set(float64(len(payload)))
```

Comment:
A few problems with this approach:
- If the AM restarts, the Grafana configuration might be there, but we won't get metrics for it until the next POST request from Grafana to update the config
- The metric doesn't change when the configuration is deleted

What if we updated this metric during the config sync for each tenant? E.g. in [`computeConfig()`](https://github.com/grafana/mimir/blob/b810cbe07175145c9c63d046df58637916ac9a46/pkg/alertmanager/multitenant.go#L701), by setting it to the `len(cfgs.Grafana.RawConfig)`.